### PR TITLE
refactor(typescript_indexer): integrate Kythe types in indexer

### PR DESCRIPTION
**Pull request by ayazhafiz**
Add Kythe types from `kythe.ts` to the indexer. In cases where custom
Kythe facts and edges are used, just cast them to a `FactName` or
`EdgeKind`.

### CODE REVIEW

**Comment by ayazhafiz** [comment]
cc @evmar

File: kythe/typescript/indexer.ts [hunk 1]
```diff
@@ -308,30 +303,36 @@ class Visitor {
...
     });
   }
 
   /** emitEdge emits a new edge entry, relating two VNames. */
-  emitEdge(source: VName, name: string, target: VName) {
+  emitEdge(source: VName, name: EdgeKind|OrdinalEdge, target: VName) {
     this.host.emit({
       source,
-      edge_kind: '/kythe/edge/' + name,
+      edge_kind: name,
```
Comment (evmar) [inline kythe/typescript/indexer.ts]: This is weird, should the parameter be 'kind'?

File: kythe/typescript/indexer.ts [hunk 2]
```diff
@@ -1106,7 +1117,7 @@ class Visitor {
     if (vname && decl.kind === ts.SyntaxKind.PropertyDeclaration) {
       const declNode = decl as ts.PropertyDeclaration;
       if (this.isStaticMember(declNode, declNode.parent)) {
-        this.emitFact(vname, 'tag/static', '');
+        this.emitFact(vname, '/kythe/tag/static' as FactName, '');
```
Comment (evmar) [inline kythe/typescript/indexer.ts]: Should this be in kythe.ts?

File: kythe/typescript/test.ts [hunk 3]
```diff
@@ -146,7 +147,7 @@ async function testPlugin() {
         };
         context.emit({
           source: pluginMod,
-          fact_name: '/kythe/node/pluginKind',
+          fact_name: '/kythe/node/pluginKind' as kythe.FactName,
```
Comment (evmar) [inline kythe/typescript/test.ts]: Should this be in kythe.ts?

File: kythe/typescript/indexer.ts (hunk 2 above)
Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: This is a non-standard Kythe fact name. We can add it though.

File: kythe/typescript/test.ts (hunk 3 above)
Comment (ayazhafiz) [reply kythe/typescript/test.ts]: This definitely shouldn't be because it is only used for testing

File: kythe/typescript/indexer.ts (hunk 1 above)
Comment (ayazhafiz) [reply kythe/typescript/indexer.ts]: yes

**Comment by ayazhafiz** [comment]
@evmar does this look good to you?

**Pull request merged**